src/installer/corehost/cli/hostpolicy/hostpolicy.cpp

```diff
@@ -410,13 +433,18 @@ SHARED_API int HOSTPOLICY_CALLTYPE corehost_main(const int argc, const pal::char
 
 SHARED_API int HOSTPOLICY_CALLTYPE corehost_main_with_output_buffer(const int argc, const pal::char_t* argv[], pal::char_t buffer[], int32_t buffer_size, int32_t* required_buffer_size)
 {
-    arguments_t args;
-    int rc = corehost_main_init(g_init, argc, argv, _X("corehost_main_with_output_buffer"), args);
+    int rc = corehost_main_init(g_init, argc, argv, _X("corehost_main_with_output_buffer"));
     if (rc != StatusCode::Success)
         return rc;
 
     if (g_init.host_command == _X("get-native-search-directories"))
     {
+        arguments_t args;
+        if (!parse_arguments(g_init, argc, argv, args))
+            return StatusCode::LibHostInvalidArgs;
+
+        args.trace();
```

Comment:
Can we move the `args.trace()` inside the `parse_arguments`? It seems that all places where we call `parse_arguments` also call `args.trace()`.